src/clj/game/cards/ice.clj

```diff
@@ -889,6 +889,39 @@
    "Galahad"
    (grail-ice end-the-run)
 
+   "Gatekeeper"
+   (let [draw {:async true
+               :prompt "Draw how many cards?"
+               :choices {:number (req 3)
+                         :max (req 3)
+                         :default (req 1)}
+               :msg (msg "draw " target "cards")
+               :effect (effect (draw eid target nil))}
+         reveal-and-shuffle {:async true
```

Comment:
I don't believe the async is necessary, which would also eliminate all the `effect-completed`s.

Reply:
Even with the prompts?